app/cdap/services/datasource/DataSourceHttp.ts

```diff
@@ -0,0 +1,460 @@
+        this.resumePoll();
+        this.pausedPolling = false;
+      }
+    });
+  }
+
+  public getBindingsForHealthCheck() {
+    const bindingsWithTime = {};
+    // TODO This will allow callers to see the latency of polling calls
+    // but not one-time requests. We should change the system delay feature
+    // so that the data source collects latency data rather
+    // then exposing request details
+    Object.values(this.polling).forEach((binding) => {
+      if (!binding.excludeFromHealthCheck) {
+        const id = objectQuery(binding, 'resource', 'id');
+        const requestTime = objectQuery(binding, 'resource', 'requestTime');
+        bindingsWithTime[id] = requestTime;
+      }
+    });
+    return bindingsWithTime;
+  }
+
+  public createResponseHandler(bindingInfo: IBinding) {
+    return (ajaxResponse) => {
+      const errorCode = objectQuery(ajaxResponse.response, 'errorCode') || null;
+      this.eventEmitter.emit(globalEvents.API_ERROR, errorCode !== null);
+      if (ajaxResponse.status > 299) {
+        /**
+         * There is an issue here. When backend goes down we stop all the poll
+         * and inspite of stopping all polling calls and unsubscribing all subscribers
+         * we still get the rx.error call which tries to set the observers length to 0
+         * and errors out. This doesn't harm us today as when system comes up we refresh
+         * the UI and everything loads.
+         *
+         * This is being wrapped in a try catch block in case the subscriber do not define
+         * an error callback. Without this, the error will bubble up as an uncaught error
+         * and terminate the socketData subscriber.
+         */
+        try {
+          bindingInfo.rx.error({
+            statusCode: ajaxResponse.statusCode,
+            response: ajaxResponse.response || ajaxResponse.body || ajaxResponse.error,
+          });
+        } catch (e) {
+          /* tslint:disable:no-console */
+          console.groupCollapsed('Error: ' + ajaxResponse.resource.url);
+          console.log('Resource', ajaxResponse.resource);
+          console.log('Error', e);
+          console.groupEnd();
+          /* tslint:enable:no-console */
+        }
+      } else {
+        debugLog(`Sending response to subscribers for ${bindingInfo.resource.id}`);
+        bindingInfo.rx.next(ajaxResponse.response);
+
+        if (bindingInfo.type === 'POLL') {
+          this.startClientPoll(bindingInfo.resource.id);
+        } else {
+          bindingInfo.rx.complete();
+        }
+      }
+    };
+  }
+
+  public request(resource: IInboundResource = {}) {
+    const excludeFromHealthCheck = !!resource.excludeFromHealthCheck;
+    const generatedResource: IResource = {
+      id: resource.id || uuidV4(),
+      json: resource.json === false ? false : true,
+      method: resource.method || 'GET',
+      suppressErrors: resource.suppressErrors || false,
+    };
+
+    if (resource.body) {
```

Comment:
I was wondering why the next file looks so familiar - disregard that comment about making a helper down there. The websockets file should be removed eventually so its only repeated once